code/modules/wiremod/components/utility/multiplexer.dm

```diff
@@ -0,0 +1,65 @@
+/**
+ * # Combiner Component
+ *
+ * Combines multiple inputs into 1 output port.
+ */
+/obj/item/circuit_component/multiplexer
+	display_name = "Multiplexer"
+	display_desc = "A component that allows you to selectively choose which input port provides an output. The first port is the selector and takes a number between 1 and the maximum port amount."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	/// The port to select from, goes from 1 to input_port_amount
+	var/datum/port/input/input_port
+
+	/// The amount of input ports to have
+	var/input_port_amount = 4
+
+	var/datum/port/output/output_port
+
+	/// Current type of the ports
+	var/current_type
+
+	/// The multiplexer inputs. These are what get selected for the output by the input_port.
+	var/list/datum/port/input/multiplexer_inputs
+
+GLOBAL_LIST_INIT(comp_multiplexer_options, list(
+	COMP_TYPE_ANY,
+	PORT_TYPE_STRING,
+	PORT_TYPE_NUMBER,
+	PORT_TYPE_LIST,
+	PORT_TYPE_ATOM
+))
+
+/obj/item/circuit_component/multiplexer/Initialize()
+	options = GLOB.comp_multiplexer_options
```

Comment:
Do you need to access this externally? Else you could just make it a static variable of the type.

Reply:
don't need to access it externally, but I got used to making it a global variable. They're all currently like this though that could change.

Reply:
```dm
/obj/item/circuit_component/multiplexer
	var/static/list/options = list("A", "B", "C")
```
This works the same, but with less steps.